smatch/matcher.py

```diff
@@ -260,28 +256,20 @@ def query_self(self, radius, min_match=1, eps=0.0, return_indices=False):
         """
         angle = 2.0*np.sin(np.deg2rad(radius)/2.0)
         idx = self.tree.query_ball_tree(self.tree, angle, eps=eps)
-        # The matching works best when sorting with the most matches first
-        len_arr = np.array([len(j) for j in idx])
-        st = np.argsort(len_arr)[:: -1]
-        match_id = np.full(len(idx), -1, dtype=np.int32)
-        idx2 = []
-        for j in st:
-            if match_id[j] < 0 and len_arr[j] >= min_match:
-                match_id[idx[j]] = j
-                idx2.append(idx[j])
```

Comment:
Why did you get rid of this section?  It's absolutely critical to remove dependence on ordering of the input catalog.

Reply:
The ordering relative to the inputs changing is a bug.